audits/trading/202404-threat-model-immutable-signed-zone-v2.md

```diff
@@ -2,14 +2,214 @@
 
 ## Introduction
 
-This document is a threat model for the [Immutable Signed Zone (v2)](../../contracts/trading/seaport/zones/immutable-signed-zone/v2/README.md) contract built by Immutable.
+This threat model document for the [Immutable Signed Zone (v2)](../../contracts/trading/seaport/zones/immutable-signed-zone/v2/README.md) contract has been created in preparation for internal audit.
+
+## Rationale
+
+Immutable operates a global off-chain orderbook across Immutable zkEVM chains and relies on the Seaport protocol for on-chain settlement. The orderbook primarily intends to:
+
+* Centralise liquidity
+* Enforce payment of fees (protocol, royalty, marketplace)
+
+The [Immutable Signed Zone (v2)](../../contracts/trading/seaport/zones/immutable-signed-zone/v2/README.md) is an implementation of [SIP-7: Server-Signed Orders [TBD]](). The reasons Immutable has chosen this solution is:
+
+* It requires orders to be known by the orderbook prior to fulfilment
+* It allows the orderbook to refuse fulfilment of an order at its discretion (e.g. for gasless cancellations, or compromised collections)
+* It allows for arbitrary logic to be defined off-chain and enforced on-chain (e.g. to enforce fees)
+
+## Threat Model Scope
+
+The threat model is limited to the following Solidity files at GitHash [TBD]:
```

Comment:
Add the githash and link to browse files at that githash. That is needed, so people can understand what the threat model was based on

Reply:
Will do once we've actioned code changes below.

Reply:
Added links to githash